pkg/client/client.go

```diff
@@ -37,6 +37,9 @@ type Options struct {
 
 	// Mapper, if provided, will be used to map GroupVersionKinds to Resources
 	Mapper meta.RESTMapper
+
+	// DryRun specifies whether dry run mode should be enforced
+	DryRun bool
```

Comment:
What's the main usecase for having this as an option here vs just letting people call `NewDryRunClient`?  Just want to avoid too many options here, esp since this isn't strictly related to creating the client

Reply:
Good point, its not really necessary there, I removed it.

I would like to keep it in the `manager` though, because the `defaultNewClient` is not exported and replacing the `NewClient` func is not very intuitive and an option ppl that are not deeply familiar with c-r will most likely not become aware of. WDYT?

Reply:
Not super excited about growing manager's Options.

One alternative we can try here is that:
Making `defaultNewClient` exportable, but under `pkg/manager/internal` or `pkg/internal/manager` or some place similar.
Then you can have a helper method that satisfy `NewClientFunc`
```
func NewDryRunClientFn(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
  cl, err := managerinternal.DefaultNewClient(cache, config, options) 
  return &dryRunClient{cl}, err
}
```
User can pass this method to manager Options.

@DirectXMan12 Thoughts?

Reply:
@mengqiy the reason I dislike that is that its pretty hard to discover that these kind of build-in opts exists